registry/storage/registry.go

```diff
@@ -12,28 +12,60 @@ import (
 // package. All instances should descend from this object.
 type registry struct {
 	blobStore                   *blobStore
-	blobServer                  distribution.BlobServer
-	statter                     distribution.BlobStatter // global statter service.
+	blobServer                  *blobServer
```

Comment:
Let's maintain the interfaces here. We can type assert in the option method.

This will avoid the temptation of interface creep.